### PR TITLE
chore: remove some unsafe

### PR DESCRIPTION
**Pull request by u2**
Is the unsafe necessary?

I tried to add `#[cfg(test)]` for `build_unchecked` of `Block`, but the compiling failed.

### CODE REVIEW

**Comment by yangby-cryptape** [comment]
- If users use the `build(..)` to build those structs, the hashes will be correct.
- If users use the `build_unchecked(..)` to build those structs, the correctness of those hashes can not be guaranteed.

So, both @driftluo and @TheWaWaR suggested that add `unsafe` to let users pay attention to those methods.

In fact, we shouldn't use these methods in non-test code. We tried using `$[cfg(test)]`, but the `$[cfg(test)]` can't passed from crates to crates. (We add a `$[cfg(test)]` in crate A, and run `cargo test` in crate B, crate A couldn't know that it should be enable the `test` feature.)

File: store/src/flat_block_body.rs [hunk 1]
```diff
@@ -137,17 +137,13 @@ pub(crate) fn deserialize_transaction(
...
-    unsafe {
-        Ok(TransactionBuilder::default()
-            .version(version)
-            .deps(deps)
-            .inputs(inputs)
-            .outputs(outputs)
-            .witnesses(witnesses)
-            .build_unchecked(hash, witness_hash))
-    }
+    Ok(TransactionBuilder::default()
```
Comment (yangby-cryptape) [inline store/src/flat_block_body.rs]: We wrote `hashes` into storage as caches. `build(..)` always computes those hashes again.

Comment (u2) [reply store/src/flat_block_body.rs]: updated.

**Comment by doitian** [comment]
It is intended to mark those functions as unsafe.

**Comment by u2** [comment]
> It is intended to mark those functions as unsafe.

The unsafe in Rust means that not enforcing the memory safety guarantees.

**Comment by doitian** [comment]
It's the best way we know to prevent using the function by accident. It is not acceptable to simply remove the unsafe.

**Pull request closed without merging**